src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @flow
+ * @file ContentSharing Element
+ * @author Box
+ */
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import type { $AxiosError } from 'axios';
+import API from '../../api';
+import Internationalize from '../common/Internationalize';
+import ErrorMask from '../../components/error-mask/ErrorMask';
+import UnifiedShareModal from '../../features/unified-share-modal';
+import usmMessages from '../../features/unified-share-modal/messages';
+import { convertItemResponse, convertUserResponse } from '../../features/unified-share-modal/utils/convertData';
+import { CLIENT_NAME_CONTENT_SHARING, FIELD_ENTERPRISE, FIELD_HOSTNAME, TYPE_FILE, TYPE_FOLDER } from '../../constants';
+import { CONTENT_SHARING_ERRORS, CONTENT_SHARING_ITEM_FIELDS } from './constants';
+import contentSharingMessages from './messages';
+import type { ErrorResponseData } from '../../common/types/api';
+import type { ItemType } from '../../common/types/core';
+import type { item as itemFlowType } from '../../features/unified-share-modal/flowTypes';
+import type { ContentSharingItemAPIResponse, ContentSharingSharedLinkType } from './types';
+
+type ContentSharingProps = {
+    apiHost: string,
+    displayInModal?: boolean,
+    itemID: string,
+    itemType: ItemType,
+    language: string,
+    token: string,
+};
+
+function ContentSharing(props: ContentSharingProps) {
+    const { apiHost, displayInModal, itemID, itemType, language, token } = props;
+    const [api, setAPI] = React.useState<API>(
+        new API({
+            apiHost,
+            clientName: CLIENT_NAME_CONTENT_SHARING,
+            id: `${itemType}_${itemID}`,
+            token,
+        }),
+    );
+    const [item, setItem] = React.useState<itemFlowType | null>(null);
+    const [sharedLink, setSharedLink] = React.useState<ContentSharingSharedLinkType | null>(null);
+    const [currentUserID, setCurrentUserID] = React.useState<string | null>(null);
+    const [errorMessage, setErrorMessage] = React.useState<Object | null>(null);
+
+    React.useEffect(() => {
+        setAPI(
+            new API({
+                apiHost,
+                clientName: CLIENT_NAME_CONTENT_SHARING,
+                id: `${itemType}_${itemID}`,
+                token,
+            }),
+        );
+    }, [apiHost, itemID, itemType, token]);
+
+    const getError = React.useCallback(
+        (error: $AxiosError<Object> | ErrorResponseData) => {
+            let errorObject;
+            if (error.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.status]];
+            } else if (error.response && error.response.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.response.status]];
+            } else {
+                errorObject = contentSharingMessages.loadingError;
+            }
+
+            setErrorMessage(errorObject);
+        },
+        [setErrorMessage],
```

Comment:
Should this be `[errorMessage]` so the callback memoization is keyed off the string rather than the setter?

Reply:
That makes more sense to me, too, but the React linter requires `setErrorMessage` (since the function references `setErrorMessage` instead of `errorMessage` directly)